### PR TITLE
Caf-2846: Environment Variable

### PR DESCRIPTION
**Pull request by arogan92**

### CODE REVIEW

File: caf-audit-service-container/README.md [hunk 1]
```diff
@@ -20,7 +20,30 @@ The number of replica shards (copies) that each primary shard should have. e.g.
 
 ##### CAF\_AUDIT\_SERVICE\_API\_CONFIG\_PATH
 
-The path to the directory containing the config.properties which can be used as an alternative means of specifying the environment variables. e.g. .\config-props
```
Comment (mulhollandc) [inline caf-audit-service-container/README.md:23]: This section should begin with CAF_LOG_LEVEL with a paragraph or set of paragraphs directly under (just like the other settings).

Does audit-service-deploy README not require a similar change?

File: caf-audit-service/src/main/resources/log4j2.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+    <Configuration>
+        <Appenders>
+            <Console name="Console" target="SYSTEM_OUT">
+                <PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
```
Comment (mulhollandc) [inline caf-audit-service/src/main/resources/log4j2.xml]: Is this the same pattern used in other services. e.g. document-id and job-service? Maybe it doesn't matter - just a thought.

**Review by mulhollandc — CHANGES_REQUESTED** [review]

File: caf-audit-service-container/README.md (hunk 1 above)
Comment (arogan92) [reply caf-audit-service-container/README.md:23]: Readme for audit-service-deploy is in a separate pull request. (https://github.com/CAFAudit/audit-service-deploy/pull/5)

I'll update this readme and push the changes

**Comment by arogan92** [comment]
Readme updated after comments from Connor

File: caf-audit-service/src/main/resources/log4j2.xml (hunk 2 above)
Comment (arogan92) [reply caf-audit-service/src/main/resources/log4j2.xml]: The others contain a package and status in the configuration header. Sitting with Andy this morning its apparent that we don't really need that.

I don't think we require to separate the Logging levels amongst different packages either? Like that of Document-id?

Comment (mulhollandc) [reply caf-audit-service/src/main/resources/log4j2.xml]: Sorry but I wasn't referring to separate logging levels etc. I meant the actual PatternLayout format you have specified.

File: caf-audit-service-container/README.md [hunk 3]
```diff
@@ -22,22 +22,11 @@ The number of replica shards (copies) that each primary shard should have. e.g.
 
 The path to the directory containing the config.properties which can be used as an alternative means of specifying the environment variables. e.g. .\config-props
 
-#### Logging
+##### CAF\_LOG\_LEVEL
```
Comment (mulhollandc) [inline caf-audit-service-container/README.md]: Why must the logging level be specified as lowercase?

Change section to:

##### CAF\_LOG\_LEVEL

The logging level for the CAF Audit Web Service. e.g debug

Logging levels supported include:

* warn
* error
* info
* trace
* debug
* all

A default logging level of **info** is applied if this environment variable is not configured.

**Review by mulhollandc — CHANGES_REQUESTED** [review]

File: caf-audit-service/src/main/resources/log4j2.xml (hunk 2 above)
Comment (arogan92) [reply caf-audit-service/src/main/resources/log4j2.xml]: Yep, the pattern layout is identical to that of documentid-service

**Comment by arogan92** [comment]
Changes made to README's and code-formatting in log4j.xml files after pull request comments

File: caf-audit-service-container/README.md [hunk 4]
```diff
@@ -24,15 +24,20 @@ The path to the directory containing the config.properties which can be used as
 
 ##### CAF\_LOG\_LEVEL
 
-The Logging within the CAF Audit Web Service has been set to a base level of INFO. This logging configuration can be overridden in order to raise or lower the log levels.
+The logging level for the CAF Audit Web Service. e.g. DEBUG
 
-The default Log Level can be overridden by setting `CAF_LOG_LEVEL` to a level of your choice and re-running the Audit Web Service
+Logging levels support include:
```
Comment (mulhollandc) [inline caf-audit-service-container/README.md]: Should this be 'Logging levels supported include:' or 'The logging levels supported are:'? The line was to indicate which suitable logging levels are supported through the configuration.

**Review by mulhollandc — CHANGES_REQUESTED** [review]

File: caf-audit-service-container/README.md (hunk 4 above)
Comment (arogan92) [reply caf-audit-service-container/README.md]: You're right. Grammar error

**Review by mulhollandc — APPROVED** [review]

**Comment by arogan92** [comment]
Grammar error has been fixed

**Comment by arogan92** [comment]
Release notes updated

File: release-notes-3.1.0.md [hunk 5]
```diff
@@ -2,6 +2,7 @@
 #### Version Number
 - [CAF-2277](https://jira.autonomy.com/browse/CAF-2277): Exposed a new https endpoint for audit web service web calls (default 8443).
 - [CAF-2693](https://jira.autonomy.com/browse/CAF-2693): Added support for Elasticsearch 'keyword' and 'text' string data types.
+- [CAF-2846](https://jira.autonomy.com/browse/CAF-2846): Added support for configuring the Audit Web Service logging levels through an environment variable
```
Comment (mulhollandc) [inline release-notes-3.1.0.md]: Adam I think this should be changed to Audit Service and not specific to the Web service as you have added this to the client side code as well if I recall?

**Review by mulhollandc — CHANGES_REQUESTED** [review]

File: release-notes-3.1.0.md (hunk 5 above)
Comment (arogan92) [reply release-notes-3.1.0.md]: done

**Review by mulhollandc — APPROVED** [review]

**Pull request merged**